docs/js-lang-overview.md

```diff
@@ -0,0 +1,74 @@
+
+In a typical database environment, direct access to external files (like reading or writing files on the server's file system) is restricted. Our implementation adheres to a trusted external routine language policy, ensuring routines cannot perform operations beyond what is normally possible for database users. Consequently, file or network I/O operations are not supported within our routines.
+
+Our system supports asynchronous JS code, but it does not work well for database routines. Since everything runs on the same thread and there is nothing to wait for asynchronously, using asynchronous code is unnecessary and not recommended.
+
+We always run JS code in strict mode, and developers cannot disable or change this setting.
+
+## Convert SQL data types to JS
+
+SQL and JS use different data types, so our implementation converts values when passing SQL parameters to JS and back. The following rules explain how these conversions work:
+
+SQL `NULL` values are converted to JS `null` values.
+
+| SQL type | JS return type | Notes |
+|-----------|----------------|--------|
+| BOOLEAN, TINYINT, SHORTINT, MEDIUMINT, INT | Number | |
+| BIGINT | Number or BigInt | Number for values [-2^53-1, 2^53-1], BigInt otherwise |
+| DECIMAL | String | |
+| FLOAT, DOUBLE | Number | |
+| BIT(k) | Number or BigInt | Number for k ≤ 53, BigInt for k > 53 |
+| TIME, DATE, TIMESTAMP, DATETIME | String | |
+| YEAR | Number | |
+| CHAR, VARCHAR, TINYTEXT, TEXT, MEDIUMTEXT, LONGTEXT | String | Fails if length exceeds 2^29 - 24 |
+| BINARY, VARBINARY, TINYBLOB, BLOB, MEDIUMBLOB, LONGBLOB | DataView | |
+| ENUM, SET | String | |
+| GEOMETRY and spatial types | DataView | |
+| JSON | Object | |
+
+When the data converts to a JS string, it automatically changes from the SQL parameter’s character set to `utf8mb4`, which JS uses.
+
+## Convert JS data types to SQL
+
+The target SQL data type defines how the system converts values. The system typically converts a JS value to one of the basic types, such as string, integer, or double, depending on the SQL data type. After the conversion, the system stores the result in the SQL parameter or return value. This step can fail if the value is too large or has an incorrect format, which will cause an error. During the process, JS strings automatically convert from `utf8mb4` to the character set of the SQL parameter.
+
+JS `null` and `undefined` values always convert to SQL `NULL` values for the target SQL type.
+
+| Target SQL data type | Rules for converting JS value | Example |
+|----------------------|----------------------------|---------|
+| Integer Numbers (BOOLEAN, TINYINT, SHORTINT, MEDIUMINT, INT, BIGINT) | JS Integers/Numbers: integers stored as-is, BIGINTs attempted as integers, others as strings. | 42 → 42 |
+| Non-integer Numbers | Converted to strings | 3.14 → "3.14" |
```

Comment:
We are almost there! I think we need to remove Non-integer Numbers and All other values lines as their contents has been moved to the first line now (which is correct). We can add examples from these lines to the first line as well.